itsthejoker/examplerepo
# Closes #3734: Audit - track downloads and views for logs

Assembla: https://app.assembla.com/spaces/smartfile/tickets/3734/details

#### What does this PR do?
Fixes the issues that required this PR to originally be reverted from prod allows audit to track information regarding logs while still saving and searching existing audit entries as expected.

#### Summary of commits:
[1991051](https://github.com/smartfile/smartfile/pull/844/commits/199105110d8af560b8d7a294c985e5bbd1656717) is the revert of the original implementation; incomplete testing and a misunderstanding of the audit serializer meant that the audit page would never load. [60641f4](https://github.com/smartfile/smartfile/pull/844/commits/60641f442041b3913500360ab5b0277a8e539f1c) fixes the serializer and removes the Django ContentType lookup in favor of creating a list of audit types at runtime, and [643a263](https://github.com/smartfile/smartfile/pull/844/commits/643a2631d1215248cb58fc4d4439dd4603b31b7c) removes the tests that were getting skipped and replaces them with an appropriate test.

#### How should this be manually tested?
Generate any type of audit logs (login, logout, rename, move, etc.) and verify that they appear and are searchable on the Audit page.

#### Any background context you want to provide?
This issue was believed to have been fixed in https://github.com/smartfile/smartfile/pull/765, but we ran into an issue with incomplete testing and the audit page not loading.

#### Are there additional relevant tickets?
N/A

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
